src/components/Cart.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect, useContext, useRef } from 'react'
+import { CartInfoContext } from "../providers/CartInfoProvider";
+
+const Cart = () => {
+    const { _confirmedOrdersInfo, _updateCartInfo } = useContext(CartInfoContext);
+
+    const [confirmedOrdersInfo, setConfirmedOrdersInfo] = _confirmedOrdersInfo;
+    const [updateCartInfo, setUpdateCartInfo] = _updateCartInfo
+
+
+    // checking and getting the items from local storage and storing them into the state context value of 'confirmedOrdersInfo' onmount
+    // this will check if there is anything in the local storage. If there are values stored in the local storage, then stored them into confirmedOrdersInfo
+    useEffect(() => {
+        if (JSON.parse(localStorage.getItem("confirmed orders") !== null)) {
```

Comment:
There are 3 different actions here that should be broken down into 3 clear steps:
```js
const _confirmedOrders  = localStorage.getItem("confirmed orders")
const confirmedOrders = JSON.parse(_confirmedOrders)
```
Why are you comparing to `null`? You can just check it is not false
```js
if (confirmedOrders) { ...
```

Reply:
The string "confirmed orders" is the title of the user's order data that is stored in local storage.  `_confirmedOrders` is the keyName that contains `[confirmedOrders, setConfirmedOrders]` from the CartInfoProvider. This state context value will hold the user's cart orders. 

And the reason why I am comparing `null` is that if there's nothing in the local storage, then `JSON.parse(localStorage.getItem("confirmed orders")` will return a `null` value.